src/main/scala/com/fulcrumgenomics/vcf/AssessPhasing.scala

```diff
@@ -0,0 +1,780 @@
+        }
+        if (isTruthVariant) {
+          metric.num_truth += 1
+          if (isTruthVariantPhased) metric.num_truth_phased += 1
+        }
+        // Less fun to compute metrics
+        if (isCallVariant && isTruthVariantPhased) {
+          metric.num_called_with_truth_phased += 1
+          if (isCallVariantPhased) metric.num_phased_with_truth_phased += 1
+        }
+        if (inCallPhaseBlock) {
+          if (isTruthVariantPhased) {
+            metric.num_truth_phased_in_called_block += 1
+            if (isCallVariantPhased) metric.num_both_phased_in_called_block += 1
+          }
+        }
+
+        // NB: only create the cigar from phased variants.
+        if (isCallVariantPhased || isTruthVariantPhased) {
+          contextsToCigar(truthPhasedVariant, callPhasedVariant)
+        }
+        else Seq.empty
+      }.toList match { // remove leading `*End`
+        case Nil            => Nil
+        case BothEnd :: xs  => xs
+        case CallEnd :: xs  => xs
+        case TruthEnd :: xs => xs
+        case xs             => xs // unreachable?
+     }
+
+    // Update the various rates...
+    metric.finalizeMetric()
+
+    Seq(BothEnd) ++ cigar ++ Seq(BothEnd)
+  }
+
+ /** Gets all the indices in the cigar where there is a short switch error.
+  *
+  * Short switches are sites where the alleles are misphased (flipped), and both upstream and downstream, either the
+  * haplotype block ends, or the next variant has the same phase.
+  */
+  def toShortSwitchErrorIndices(cigar: Cigar): Seq[Int] = {
+    // check if we can find a match or the end of a block at the given idx.  If we cannot, move to the next idx.
+    @tailrec
+    def checkNext(cigar: Seq[CigarType], idx: Int, by: Int): Boolean = {
+      if (idx < 0 || cigar.length <= idx) return true // no more
+      cigar(idx) match {
+        case Match | CallEnd | BothEnd | TruthEnd => true
+        case Mismatch     => false
+        case Call | Truth => checkNext(cigar=cigar, idx=idx+by, by=by)
+      }
+    }
+    // find cigars that have a mismatch at the index, but then anchored on either side
+    cigar.indices.filter { i => cigar(i) == Mismatch && checkNext(cigar, i-1, -1) && checkNext(cigar, i+1, 1) }.toList
+  }
+
+  /** Gets the number of long switch errors and the number of sites examined.
+    *
+    * The number of sites examined are places where there are both a truth and called variant.
+    *
+    * Long switches are a stretch of two or more consecutive sites where the alleles are misphased (flipped), and
+    * both upstream and downstream, either the haplotype block ends, or the next variant has the same phase.
+    */
+  def toLongSwitchErrorsAndSites(cigar: Cigar): (Int, Int) = {
+    // check if we can find a match or the end of a block at the given idx.  If we cannot, move to the next idx.
+    @tailrec
+    def checkNext(cigar: Seq[CigarType], idx: Int, by: Int, stretchLength: Int = 0): Int = {
```

Comment:
Since this does more than check the next thing, maybe rename it to `numberOfContiguousMismatches` or something like that?